packages/contracts/test/exchange/match_orders.ts

```diff
@@ -1168,14 +1168,22 @@ describe('matchOrders', () => {
                 makerAssetAmount: Web3Wrapper.toBaseUnitAmount(new BigNumber(10), 18),
                 takerAssetAmount: Web3Wrapper.toBaseUnitAmount(new BigNumber(2), 18),
             });
+            const signedOrderRightPrim = {
+                ...signedOrderRight,
+                takerAssetData: signedOrderLeft.makerAssetData,
+            };
+            const orderHashRightPrim = orderHashUtils.getOrderHashHex(signedOrderRightPrim);
             // Match orders
-            return expectTransactionFailedAsync(
+            return expectTransactionFailedWithParamsAsync(
                 exchangeWrapper.matchOrdersAsync(signedOrderLeft, signedOrderRight, takerAddress),
                 // We are assuming assetData fields of the right order are the
                 // reverse of the left order, rather than checking equality. This
                 // saves a bunch of gas, but as a result if the assetData fields are
                 // off then the failure ends up happening at signature validation
-                RevertReason.InvalidOrderSignature,
+                {
+                    reason: RevertReason.InvalidOrderSignature,
+                    params: { orderHash: orderHashRightPrim },
+                },
```

Comment:
Nice, this is a way more precise assert then what we had about this. Feel these error messages are going to greatly improve the quality of our tests as well 🚀

Reply:
If you have a better naming idea than _Prim I'm happy to adapt it. It's just a placeholder while I'm thinking about a better one.

Reply:
I've meant `Prime` as [here](https://en.wikipedia.org/wiki/Prime_(symbol)).
_In mathematics, the prime is generally used to generate more variable names for things which are similar, without resorting to subscripts – `x′` generally means something related to or derived from `x`._

It's not the hash of the optimized order. it's the hash of the order that we get when we try to deoptimize the order on-chain assuming the optimization was done correctly. So if it was - that's the hash of the right order. If it wasn't - it's a hash of the fake order that we've never seen before even during optimization phase.

Reply:
gotcha